File: QytetetJava/src/modeloqytetet/PruebaQytetet.java

package modeloqytetet;

import java.util.ArrayList;

public class PruebaQytetet {

    private static ArrayList<Sorpresa> mazo = new ArrayList();
    private static Tablero tablero = new Tablero();
    
    private ArrayList<Sorpresa> sorpresasMayor0 () {
        ArrayList<Sorpresa> res = new ArrayList();
        
        for (int i = 0; i < mazo.size(); i++) {
            if (mazo.get(i).getValor() > 0)
                res.add(mazo.get(i));
        }
        
        return res;
    }
    
    private ArrayList<Sorpresa> IRACASILLA () {
        ArrayList <Sorpresa> res = new ArrayList();
        
        for (int i = 0; i < mazo.size(); i++) {
            if (mazo.get(i).getTipo() == TipoSorpresa.IRACASILLA)
                res.add(mazo.get(i));
        }
        
        return res;
    }
    
    private ArrayList<Sorpresa> TipoSorpresa(TipoSorpresa t) {
        ArrayList <Sorpresa> res = new ArrayList();
        
        for (int i = 0; i < mazo.size(); i++) {
            if (mazo.get(i).getTipo() == t)
                res.add(mazo.get(i));
        }
        
        return res;
    }
    
    /*private static ArrayList<Casilla> todoTablero() {
        return tablero.casillas;
    }*/
    
    private static void inicializarSorpresas() {
       mazo.add(new Sorpresa("Te hemos pillado con chanclas y calcetines, lo  sentimos, ¡debes ir a la carcel!", 9, TipoSorpresa.IRACASILLA));
       mazo.add(new Sorpresa("Pagas por la sorpresa",-1500,TipoSorpresa.PAGARCOBRAR));
       mazo.add(new Sorpresa("Ganas por la sorpresa",3000,TipoSorpresa.PAGARCOBRAR));
       mazo.add(new Sorpresa("Vas a la casilla porque...",5,TipoSorpresa.IRACASILLA));
       mazo.add(new Sorpresa("Vas a la carcel por corrupto",9,TipoSorpresa.IRACASILLA));
       mazo.add(new Sorpresa("Pagas por tener tanto y ganar tan poco",2000,TipoSorpresa.PORCASAHOTEL));
       mazo.add(new Sorpresa("A pagar impuestos",1000,TipoSorpresa.PORCASAHOTEL));
       mazo.add(new Sorpresa("Tus amigos estan generosos",300,TipoSorpresa.PORJUGADOR));
       mazo.add(new Sorpresa("Le debes pasta a tus colegas",-250,TipoSorpresa.PORJUGADOR));
       mazo.add(new Sorpresa("Vas a la casilla porque...",15,TipoSorpresa.IRACASILLA));
       mazo.add(new Sorpresa("Una pena que te vallas, sigue asi y volveras",0,TipoSorpresa.SALIRCARCEL));
    }
    
    /*public Casilla getCarcel() {
                
        for (Casilla c : tablero.casillas)
            if (c.getTipo() == TipoCasilla.CARCEL)
                return c;
        return null;
    }*/
    
    
    public static void main(String[] args) {
        inicializarSorpresas();
       // ArrayList<Casilla> tab = todoTablero();
        
        System.out.println("Vamos a ver el mazo de sorpresas:");
        System.out.println(mazo.toString());
        
        System.out.println("Vamos a ver el tablero: ");
        System.out.println(tablero.toString());
        
        System.out.println("Número de la casilla de la cárcel:");
        System.out.println(tablero.getCarcel().getNumeroCasilla());
    }
    
}

File: QytetetRuby/lib/prueba_qytetet.rb
#encoding: utf-8
require_relative "sorpresa"
require_relative "tipo_sorpresa"
require_relative "tablero"
require_relative "casilla"

module ModeloQytetet
  class PruebaQytetet
    @@mazo = Array.new
    
    
    def sorpresas_mayor0
      res = Array.new;
      
      @@mazo.each { |s| 
        if s.valor > 0 then
          res << s
        end
        
      }
      return res
    end
    
    def ir_a_casilla
      res = Array.new
        
      @@mazo.each { |s|
        if s.tipo == [TipoSorpresa::IRACASILLA] then
          res << s
        end
      }

      return res
    end
    
    def tipo_sorpresa(t)
      res = Array.new
        
      @@mazo.each { |s|  
        if s.tipo == t then
          res << s
        end
      }
  
      return res
    end
    
    def inicializar_sorpresas
      @@mazo = []
      t = Tablero.new
      @@mazo<< Sorpresa.new("Te hemos pillado con chanclas y calcetines, lo  sentimos, debes ir a la carcel", 9, [TipoSorpresa::IRACASILLA])
      @@mazo<< Sorpresa.new("Pagas por la sorpresa",-1500, [TipoSorpresa::PAGARCOBRAR])
      @@mazo<< Sorpresa.new("Ganas por la sorpresa",3000, [TipoSorpresa::PAGARCOBRAR])
      @@mazo<< Sorpresa.new("Vas a la casilla porque...",5, [TipoSorpresa::IRACASILLA])
      @@mazo<< Sorpresa.new("Vas a la carcel por corrupto", t.get_carcel.get_numero_casilla, [TipoSorpresa::IRACASILLA])
      @@mazo<< Sorpresa.new("Pagas por tener tanto y ganar tan poco",2000, [TipoSorpresa::PORCASAHOTEL])
      @@mazo<< Sorpresa.new("A pagar impuestos",1000, [TipoSorpresa::PORCASAHOTEL])
      @@mazo<< Sorpresa.new("Tus amigos estan generosos",300, [TipoSorpresa::PORJUGADOR])
      @@mazo<< Sorpresa.new("Le debes pasta a tus colegas",-250, [TipoSorpresa::PORJUGADOR])
      @@mazo<< Sorpresa.new("Vas a la casilla porque...",15, [TipoSorpresa::IRACASILLA])
      @@mazo<< Sorpresa.new("Una pena que te vallas, sigue asi y volveras",0, [TipoSorpresa::SALIRCARCEL])
    end
  
  
  def self.main
    prueba = PruebaQytetet.new
    
    prueba.inicializar_sorpresas
    
    sorpresas_casilla = prueba.ir_a_casilla
    mayor0 = prueba.sorpresas_mayor0
    salir_carcel = prueba.tipo_sorpresa([TipoSorpresa::SALIRCARCEL])
    
#    puts "ir a casilla"
#    puts sorpresas_casilla
#    puts "mayor que 0"
#    puts mayor0
    #puts "salir de la carcel"
    #puts salir_carcel
    
    t = Tablero.new
    t.inicializar
    
    #muestro las casillas de todo el tablero
    #puts t.to_s
    
    v = Array.new
    v << 3
    v << 5
    v << 9
    
    puts "accedo al contenido en la posición 2:"
    
    puts v[2]
    
    puts "Accedo a la posición de la cárcel:"
    
    puts t.get_carcel.get_numero_casilla
  end
end

  PruebaQytetet.main
end
  

